src/libpriv/rpmostree-util.c

```diff
@@ -529,3 +529,74 @@ rpmostree_str_replace (const char  *buf,
 
   return g_regex_replace_literal (regex, buf, -1, 0, new, 0, error);
 }
+
+/* Like realpath(3), but ensures we stay within the given rootfs. We assume the
```

Comment:
Basically while I would definitely believe you've written a correct version of "relative realpath" I'd have to think fairly hard to review it :smile: